File: server/Data/Entities/KnockoutPosition.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using SolidTradeServer.Data.Entities.Common;

namespace SolidTradeServer.Data.Entities
{
    public class KnockoutPosition : BaseEntity
    {
        [Required]
        public Portfolio Portfolio { get; set; }

        [Required]
        [MaxLength(12)]
        public string Isin { get; set; }
        
        public int NumberOfShares { get; set; }
        public float BuyInPrice { get; set; }
    }
}

File: client/lib/data/entities/knockout_position.dart
import 'package:solidtrade/data/entities/base/base_entity.dart';

class KnockoutPosition implements IBaseEntity {
  @override
  final int id;
  @override
  final DateTime createdAt;
  @override
  final DateTime updatedAt;

  final String isin;
  final double buyInPrice;
  final double numberOfShares;

  const KnockoutPosition({
    required this.id,
    required this.createdAt,
    required this.updatedAt,
    required this.isin,
    required this.buyInPrice,
    required this.numberOfShares,
  });

  factory KnockoutPosition.fromJson(Map<String, dynamic> json) {
    return KnockoutPosition(
      id: json["id"],
      createdAt: DateTime.parse(json["createdAt"]),
      updatedAt: DateTime.parse(json["updatedAt"]),
      isin: json["isin"],
      buyInPrice: json["buyInPrice"],
      numberOfShares: json["numberOfShares"],
    );
  }
}
